src/Microsoft.AspNet.Mvc.Core/ActionResults/JsonResult.cs

```diff
@@ -36,7 +37,18 @@ public JsonResult(object value)
         }
 
         /// <summary>
-        /// Creates a new <see cref="JsonResult"/> with the given <paramref name="data"/>.
+        /// Creates a new <see cref="JsonResult"/> with the given <paramref name="value"/>.
+        /// </summary>
+        /// <param name="value">The value to format as JSON.</param>
+        /// <param name="serializerSettings">The <see cref="JsonSerializerSettings"/> to be used by
+        /// the formatter.</param>
+        public JsonResult(object value, [NotNull] JsonSerializerSettings serializerSettings)
+            : this(value, formatter: new JsonOutputFormatter { SerializerSettings = serializerSettings })
```

Comment:
I added a constructor to JsonOutputFormatter that takes a JsonSerializerSettings argument in PR https://github.com/aspnet/Mvc/pull/2530

If it is accepted then this should be changed to use the new ctor.